src/core/server/plugins/plugins_service.ts

```diff
@@ -82,17 +93,20 @@ export class PluginsService implements CoreService<PluginsServiceSetup, PluginsS
 
     const config = await this.config$.pipe(first()).toPromise();
 
+    let contracts = new Map<PluginName, unknown>();
     if (!config.initialize || this.coreContext.env.isDevClusterMaster) {
       this.log.info('Plugin initialization disabled.');
-      return {
-        contracts: new Map(),
-        uiPlugins: this.pluginsSystem.uiPlugins(),
-      };
+    } else {
+      contracts = await this.pluginsSystem.setupPlugins(deps);
     }
 
+    const uiPlugins = this.pluginsSystem.uiPlugins();
     return {
-      contracts: await this.pluginsSystem.setupPlugins(deps),
-      uiPlugins: this.pluginsSystem.uiPlugins(),
+      contracts,
+      uiPlugins: {
+        ...uiPlugins,
+        config: this.generateUiPluginsConfigs(uiPlugins.public),
+      },
```

Comment:
uiPlugins are directly serialized in injected metadatas, so I added the configuration in a distinct property instead of trying to merge in the uiPlugins structs